github/enterprise_rules.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2025 The go-github AUTHORS. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package github
+
+import (
+	"context"
+	"fmt"
+)
+
+// CreateEnterpriseRuleset creates a ruleset for the specified enterprise.
+//
+// GitHub API docs: https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/rules#create-an-enterprise-repository-ruleset
+//
+//meta:operation POST /enterprises/{enterprise}/rulesets
+func (s *EnterpriseService) CreateEnterpriseRuleset(ctx context.Context, enterprise string, rs *Ruleset) (*Ruleset, *Response, error) {
+	u := fmt.Sprintf("enterprises/%v/rulesets", enterprise)
+
+	req, err := s.client.NewRequest("POST", u, rs)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ruleset *Ruleset
+	resp, err := s.client.Do(ctx, req, &ruleset)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return ruleset, resp, nil
+}
+
+// GetEnterpriseRuleset gets a ruleset from the specified enterprise.
+//
+// GitHub API docs: https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/rules#get-an-enterprise-repository-ruleset
+//
+//meta:operation GET /enterprises/{enterprise}/rulesets/{ruleset_id}
+func (s *EnterpriseService) GetEnterpriseRuleset(ctx context.Context, enterprise string, rulesetID int64) (*Ruleset, *Response, error) {
+	u := fmt.Sprintf("enterprises/%v/rulesets/%v", enterprise, rulesetID)
+
+	req, err := s.client.NewRequest("GET", u, nil)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var ruleset *Ruleset
+	resp, err := s.client.Do(ctx, req, &ruleset)
+	if err != nil {
+		return nil, resp, err
+	}
+
+	return ruleset, resp, nil
+}
+
+// UpdateEnterpriseRuleset updates a ruleset from the specified enterprise.
+//
+// GitHub API docs: https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/rules#update-an-enterprise-repository-ruleset
+//
+//meta:operation PUT /enterprises/{enterprise}/rulesets/{ruleset_id}
+func (s *EnterpriseService) UpdateEnterpriseRuleset(ctx context.Context, enterprise string, rulesetID int64, rs *Ruleset) (*Ruleset, *Response, error) {
```

Comment:
```suggestion
func (s *EnterpriseService) UpdateEnterpriseRuleset(ctx context.Context, enterprise string, rulesetID int64, rs Ruleset) (*Ruleset, *Response, error) {
```

Reply:
I can't see this in the lint output or in the code (this pattern was copied from the org version)?

Reply:
Oh, sorry, I forgot to explain this one.  The lint output didn't show this change.
I was suggesting this change since the `rs` (in fact, maybe we should actually call it `ruleset` here instead of `rs`) parameter is required, so I removed the pointer, showing that it is required and cannot be `nil`.

Reply:
Do you want this done for all of the ruleset methods then? As I said, I copied them from the existing org implementation which AFAIK were copied from the repo ones.

Reply:
> Do you want this done for all of the ruleset methods then? As I said, I copied them from the existing org implementation which AFAIK were copied from the repo ones.

Ah!  Bummer.  It could be that I didn't catch the other ones when they were being written.

I believe we try to pass-by-value in this repo when args are required.
We also try to be consistent... but obviously I failed here.
We also try to not create breaking API changes, but I think we value consistency over breaking the API.

So, in my opinion, I think it is valuable to pass required args as value, and I think it is valuable to be consistent... so I'm leaning toward a "yes" but am open to discussion.

Reply:
It might be worth checking which pattern is used the most? The first one I checked where a struct was being used (`EditMilestone`) was also using a pointer.

Reply:
To be clear, you may do the analysis if you desire to do so.

My preference is to pass-by-value for required structs so that it is clear that `nil` is not an acceptable parameter.

In those cases where you find pass-by-reference parameters, you will also have to check if `nil` is an acceptable parameter.

Meanwhile, my request stands, both for this endpoint and for `CreateEnterpriseRuleset` above, where `nil` is definitely _not_ acceptable.

Reply:
Sorry, I thought I'd already fixed the cases in the new code (I have now). My question was about if there needs to be a breaking change for consistency wouldn't it be best to do the whole library in one go?

Reply:
I've updated the other rule implementations for consistency.